src/store/traction/mutations.js

```diff
@@ -2,6 +2,12 @@ const mutations = {
     setTags(state, tags) {
         state.tractionTags = tags
     },
+    updateRun(state, run) {
+        let runIndex = state.pacbio.runs.runs.map(e => e.id).indexOf(run.id)
+
+        state.pacbio.runs.runs.splice(runIndex, 1, run)
+        state.pacbio.runs.runs = [...state.pacbio.runs.runs]
```

Comment:
I was also wondering what is going on here?

Reply:
I noticed when I was working on the 'addLibraryToWell' mutation that the speciality feature of vuex, being that it automatically reloads components etc, wasnt working properly and this was becuase we were mutating the vuex state, so the refresh wasnt being trigger. 
The State should be immutable, so the last line there is creating a new object with the old objects state triggering the refresh. So we dont need to recall the provider() functions. 
I think we should try and look throughout the application to make sure we are using Vuex in the vuex way. In that the component calls the action, which then commits a mutation, which updates the state immutably.  
![](https://vuex.vuejs.org/vuex.png)

Hope that makes sense!!